From 2ab0d90165b150acd723ab15c144995b94eb5db7 Mon Sep 17 00:00:00 2001
From: artas182x <artas182x@gmail.com>
Date: Mon, 14 Jul 2014 21:43:37 +0200
Subject: [PATCH 099/100] Bypass fails of applying voltage. I don;t know why
 minuv of some regulators gets strange values. It is little workaround for
 this

---
 drivers/regulator/core.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/regulator/core.c b/drivers/regulator/core.c
index d3e3879..b827608 100644
--- a/drivers/regulator/core.c
+++ b/drivers/regulator/core.c
@@ -831,8 +831,9 @@ static int machine_constraints_voltage(struct regulator_dev *rdev,
 
 		/* final: [min_uV..max_uV] valid iff constraints valid */
 		if (max_uV < min_uV) {
-			rdev_err(rdev, "unsupportable voltage constraints\n");
-			return -EINVAL;
+			rdev_err(rdev, " [artas182x] Unsupportable voltage constraints. Forcing applying\n");
+			constraints->max_uV = constraints->min_uV;
+			max_uV = min_uV;
 		}
 
 		/* use regulator's subset of machine constraints */
-- 
1.9.1

